pkgdb/src/registry/floxpkgs.cc

```diff
@@ -0,0 +1,696 @@
+                         : "https://%s/api/v3/repos/%s/%s/commits/%s",
+                       host,
+                       getOwner( input ),
+                       getRepo( input ),
+                       *input.getRef() );
+
+  nix::Headers headers = makeHeadersWithAuthTokens( host );
+
+  auto json = nlohmann::json::parse( nix::readFile( store->toRealPath(
+    nix::fetchers::downloadFile( store, url, "source", false, headers )
+      .storePath ) ) );
+  auto rev  = nix::Hash::parseAny( std::string { json["sha"] }, nix::htSHA1 );
+  // nix::debug( "HEAD revision for '%s' is %s", url, rev.gitRev() );
+  return rev;
+}
+
+DownloadUrl
+GitHubInputScheme::getDownloadUrl( const nix::fetchers::Input & input ) const
+{
+  auto host = getHost( input );
+
+  nix::Headers headers = makeHeadersWithAuthTokens( host );
+
+  // If we have no auth headers then we default to the public archive
+  // urls so we do not run into rate limits.
+  const auto urlFmt
+    = host != "github.com" ? "https://%s/api/v3/repos/%s/%s/tarball/%s"
+      : headers.empty()    ? "https://%s/%s/%s/archive/%s.tar.gz"
+                           : "https://api.%s/repos/%s/%s/tarball/%s";
+
+  const auto url = nix::fmt( urlFmt,
+                             host,
+                             getOwner( input ),
+                             getRepo( input ),
+                             input.getRev()->to_string( nix::Base16, false ) );
+
+  return DownloadUrl { url, headers };
+}
+
+void
+GitHubInputScheme::clone( const nix::fetchers::Input & input,
+                          const nix::Path &            destDir ) const
+{
+  auto host = getHost( input );
+  nix::fetchers::Input::fromURL( nix::fmt( "git+https://%s/%s/%s.git",
+                                           host,
+                                           getOwner( input ),
+                                           getRepo( input ) ) )
+    .applyOverrides( input.getRef(), input.getRev() )
+    .clone( destDir );
+}
+
+/* -------------------------------------------------------------------------- */
+
+
+std::string
+FloxFlakeScheme::type() const
+{
+  return flox::FLOX_FLAKE_TYPE;
+}
+
+std::optional<nix::fetchers::Input>
+FloxFlakeScheme::inputFromURL( const nix::ParsedURL & url ) const
+{
+  /* TODO: if the type is flox-nixpkgs we can short circuit this */
+  /* don't try to convert github references */
+  if ( url.scheme != this->type() ) { return {}; }
+  auto asGithub   = url;
+  asGithub.scheme = "github";
+  GitHubInputScheme                   githubScheme;
+  std::optional<nix::fetchers::Input> fromGithub
+    = githubScheme.inputFromURL( asGithub );
+  if ( fromGithub.has_value() )
+    {
+      auto     input = *fromGithub;
+      OurAttrs attrs( getRulesHash(), getRulesProcessorHash() );
+      input.attrs  = fromGitHubAttrs( input.attrs, attrs );
+      input.scheme = std::make_shared<FloxFlakeScheme>();
+      return input;
+    }
+  else { return {}; }
```

Comment:
suggestion/question: does this really need to be github specific or can we parse `flox+<flakeref>` and parse the rest?